mtags/src/main/scala-3/scala/meta/internal/pc/completions/AmmoniteFileCompletions.scala

```diff
@@ -0,0 +1,127 @@
+import dotty.tools.dotc.core.Names.Name
+import dotty.tools.dotc.core.StdNames
+import dotty.tools.dotc.core.SymDenotations.SymDenotation
+import dotty.tools.dotc.core.Symbols.NoSymbol
+import dotty.tools.dotc.core.Symbols.Symbol
+import dotty.tools.dotc.core.Types.Type
+import dotty.tools.dotc.interactive.Interactive
+import dotty.tools.dotc.interactive.InteractiveDriver
+import dotty.tools.dotc.util.SourceFile
+import dotty.tools.dotc.util.SourcePosition
+import dotty.tools.dotc.util.Spans.Span
+import org.eclipse.{lsp4j as l}
+
+object AmmoniteFileCompletions:
+  val CURSOR = "_CURSOR_"
+
+  private def translateImportToPath(tree: Tree): String =
+    tree match
+      case Select(qual, name) =>
+        val pathPart = name.toString()
+        translateImportToPath(qual) + "/" + {
+          if pathPart == "$up" then ".."
+          else pathPart
+        }
+      case Ident(_) =>
+        ""
+      case _ => ""
+
+  def contribute(
+      select: Tree,
+      selector: List[ImportSelector],
+      editRange: l.Range,
+      rawPath: String,
+      workspace: Option[Path],
+      rawFileName: String,
+  ): List[CompletionValue] =
+    try
+      val fileName = rawFileName
+        .split("/")
+        .last
+        .stripSuffix(".amm.sc.scala")
+
+      val split = rawPath
+        .split("\\$file")
+        .toList
+
+      val query = selector.collectFirst {
+        case sel: ImportSelector if sel.name.toString().contains(CURSOR) =>
+          sel.name.toString.replace(CURSOR, "")
+
+        case sel: ImportSelector =>
+          sel.name.toString
+      }
+
+      def parent =
+        val name = "^"
+
+        CompletionValue.FileSystemMember(
+          isDirectory = true,
+          name,
+          editRange,
+        )
+
+      (split, workspace) match
+        case (_ :: script :: Nil, Some(workspace)) =>
+          // drop / or \
+          val current = workspace.resolve(script.drop(1))
+          val importPath = translateImportToPath(select).drop(1)
+          val currentPath = AbsolutePath(
+            current.getParent.resolve(importPath)
+          )
+          val parentTextEdit =
+            if query
+                .exists(
+                  _.isEmpty()
+                ) && currentPath.parentOpt.isDefined && currentPath.isDirectory
+            then List(parent)
+            else Nil
+          currentPath.list.toList
+            .filter(_.filename.stripSuffix(".sc") != fileName)
+            .collect {
+              case file
+                  if (file.isDirectory || file.isAmmoniteScript) && query
+                    .exists(
+                      CompletionFuzzy.matches(_, file.filename)
+                    ) =>
+                CompletionValue.FileSystemMember(
+                  isDirectory = file.isDirectory,
+                  file.filename,
+                  editRange,
+                )
+            } ++ parentTextEdit
+        case _ =>
+          Nil
+      end match
+    catch
```

Comment:
No need to catch it here, it should be caught in the CompilerAccess and shown properly.